packages/ra-ui-materialui/src/button/DeleteButton.js

```diff
@@ -26,44 +30,38 @@ const styles = theme => ({
 });
 
 class DeleteButton extends Component {
-    state = { dialogOpen: false };
```

Comment:
We need two buttons actually: one with the Confirm dialog, and one with the undo snackbar. Scratching my head about how to achieve that.